### PR TITLE
Set "--kubelet-preferred-address-types" if ssh tunnel is not to be used.

### PR DESCRIPTION
**Pull request by dnardo**
In additional don't advertise external address and allow the internal address
to be advertised.

**What this PR does / why we need it**:

**Which issue this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close that issue when PR gets merged)*: fixes #

**Special notes for your reviewer**:

**Release note**:

```release-note
```

### CODE REVIEW

**Review by MrHohn — APPROVED** [review]
LGTM, could you make an equivalent change to https://github.com/kubernetes/kubernetes/blob/master/cluster/gce/container-linux/configure-helper.sh as well?

**Comment by dnardo** [comment]
@MrHohn Done

**Comment by MrHohn** [comment]
Thanks!

/lgtm

**Comment by dnardo** [comment]
@bowei ping.

File: cluster/gce/container-linux/configure-helper.sh [hunk 1]
```diff
@@ -964,9 +964,13 @@ function start-kube-apiserver {
   fi
   if [[ -n "${PROJECT_ID:-}" && -n "${TOKEN_URL:-}" && -n "${TOKEN_BODY:-}" && -n "${NODE_NETWORK:-}" ]]; then
     local -r vm_external_ip=$(curl --retry 5 --retry-delay 3 --fail --silent -H 'Metadata-Flavor: Google' "http://metadata/computeMetadata/v1/instance/network-interfaces/0/access-configs/0/external-ip")
-    params+=" --advertise-address=${vm_external_ip}"
-    params+=" --ssh-user=${PROXY_SSH_USER}"
-    params+=" --ssh-keyfile=/etc/srv/sshproxy/.sshkeyfile"
+    if [[ -n "${PROXY_SSH_USER:-}" ]]; then
+            params+=" --advertise-address=${vm_external_ip}"
```
Comment (bowei) [inline cluster/gce/container-linux/configure-helper.sh]: indentation is a little strange here

File: cluster/gce/gci/configure-helper.sh [hunk 2]
```diff
@@ -1423,9 +1423,13 @@ function start-kube-apiserver {
   fi
   if [[ -n "${PROJECT_ID:-}" && -n "${TOKEN_URL:-}" && -n "${TOKEN_BODY:-}" && -n "${NODE_NETWORK:-}" ]]; then
     local -r vm_external_ip=$(curl --retry 5 --retry-delay 3 --fail --silent -H 'Metadata-Flavor: Google' "http://metadata/computeMetadata/v1/instance/network-interfaces/0/access-configs/0/external-ip")
-    params+=" --advertise-address=${vm_external_ip}"
-    params+=" --ssh-user=${PROXY_SSH_USER}"
-    params+=" --ssh-keyfile=/etc/srv/sshproxy/.sshkeyfile"
+    if [[ -n "${PROXY_SSH_USER:-}" ]]; then
+            params+=" --advertise-address=${vm_external_ip}"
```
Comment (bowei) [inline cluster/gce/gci/configure-helper.sh]: same

File: cluster/gce/container-linux/configure-helper.sh (hunk 1 above)
Comment (dnardo) [reply cluster/gce/container-linux/configure-helper.sh]: Indeed fixed.

**Pull request merged**